pydantic_ai_slim/pydantic_ai/profiles/google.py

```diff
@@ -1,106 +1,173 @@
     )
 
 
-class GoogleJsonSchemaTransformer(JsonSchemaTransformer):
-    """Transforms the JSON Schema from Pydantic to be suitable for Gemini.
+def google_vertex_model_profile(model_name: str) -> ModelProfile | None:
+    """Get the model profile for a Google Vertex AI model.
+
+    Vertex AI supports enhanced JSON Schema features as of November 2025.
+    """
+    is_image_model = 'image' in model_name
+    return ModelProfile(
+        json_schema_transformer=GoogleVertexJsonSchemaTransformer,
+        supports_image_output=is_image_model,
+        supports_json_schema_output=not is_image_model,
+        supports_json_object_output=not is_image_model,
+        supports_tools=not is_image_model,
+    )
 
-    Gemini which [supports](https://ai.google.dev/gemini-api/docs/function-calling#function_declarations)
-    a subset of OpenAPI v3.0.3.
 
-    Specifically:
-    * gemini doesn't allow the `title` keyword to be set
-    * gemini doesn't allow `$defs` — we need to inline the definitions where possible
+def google_gla_model_profile(model_name: str) -> ModelProfile | None:
+    """Get the model profile for a Google Generative Language API model.
+
+    GLA has more limited JSON Schema support compared to Vertex AI.
+    """
+    is_image_model = 'image' in model_name
+    return ModelProfile(
+        json_schema_transformer=GoogleGLAJsonSchemaTransformer,
+        supports_image_output=is_image_model,
+        supports_json_schema_output=not is_image_model,
+        supports_json_object_output=not is_image_model,
+        supports_tools=not is_image_model,
+    )
+
+
+class GoogleVertexJsonSchemaTransformer(JsonSchemaTransformer):
+    """Transforms the JSON Schema from Pydantic to be suitable for Gemini via Vertex AI.
+
+    Gemini supports [a subset of OpenAPI v3.0.3](https://ai.google.dev/gemini-api/docs/function-calling#function_declarations).
+
+    As of November 2025, Gemini 2.5+ models via Vertex AI support enhanced JSON Schema features
+    (see [announcement](https://blog.google/technology/developers/gemini-api-structured-outputs/)) including:
+    * `title` for short property descriptions
+    * `anyOf` and `oneOf` for conditional structures (unions)
+    * `$ref` and `$defs` for recursive schemas and reusable definitions
+    * `minimum` and `maximum` for numeric constraints
+    * `additionalProperties` for dictionaries
+    * `type: 'null'` for optional fields
+    * `prefixItems` for tuple-like arrays
+
+    Not supported (empirically tested as of November 2025):
+    * `exclusiveMinimum` and `exclusiveMaximum` are not yet supported by the Google SDK
+    * `discriminator` field causes validation errors with nested oneOf schemas
     """
 
     def __init__(self, schema: JsonSchema, *, strict: bool | None = None):
-        super().__init__(schema, strict=strict, prefer_inlined_defs=True, simplify_nullable_unions=True)
+        super().__init__(schema, strict=strict, prefer_inlined_defs=False, simplify_nullable_unions=True)
```

Comment:
- We can drop `prefer_inlined_defs` as it defaults to `False`
- Do we still need `simplify_nullable_unions`? `type: 'null'` is now supported natively